src/services/oracleDatapoint.ts

```diff
@@ -0,0 +1,351 @@
+import { Pool } from "pg";
+
+import { Request, Response } from "express";
+
+export interface Datapoint {
+  [address: string]: {
+    blockNumber: number;
+    blockDistance: number | null; // null if block parameter is not used
+    txHash: string;
+    txIndex: number;
+    payload: JSON;
+  };
+}
+
+interface Dictionary<T> {
+  [addresses: string]: T;
+}
+
+export const handleOracleDatapoint =
+  (p: Pool) =>
+  async (req: Request, res: Response): Promise<void> => {
+    const addresses = req.body.addresses; // list of trusted oracles
+    const ticker = req.body.ticker; // data point
+    const block = req.body.block; // block to which we should find the nearest data, in case of a draw we display newer
+    const source = req.body.source; // payload source
+    const count = req.body.count; // number of total closest/most recent results, not per oracle. If you want per oracle results, query a single oracle address
+
+    if (!addresses || addresses.length === 0) {
+      throw new Error("Addresses of trusted oracles is missing");
+    }
+
+    const ret: Dictionary<Datapoint[]> = {};
+
+    const queryMetadataOracle = `
```

Comment:
Extract SQL from the handler function into constants please, see for example:

https://github.com/Emurgo/yoroi-graphql-migration-backend/blob/master/src/services/txStatus.ts#L4-L11